reana_db/models.py

```diff
@@ -286,6 +286,58 @@ class JobStatus(enum.Enum):
     queued = 5
 
 
+class UserWorkflowSession(Base, Timestamp):
+    """User Workflow Session table."""
+
+    __tablename__ = "user_workflow_session"
+    __table_args__ = {"schema": "__reana"}
+
+    user_id = Column(UUIDType, ForeignKey("__reana.user_.id_"), nullable=False)
+    workflow_id = Column(UUIDType, ForeignKey("__reana.workflow.id_"), nullable=True)
+    session_id = Column(
+        UUIDType, ForeignKey("__reana.interactive_session.id_"), primary_key=True
+    )
+
+    def __repr__(self):
+        """User Workflow Session string representation."""
+        return "<UserWorkflowSession {} {} {}>".format(
+            self.user_id, self.session_id, self.workflow_id
+        )
+
+
+class InteractiveSessionType(enum.Enum):
+    """Enumeration of interactive session types."""
+
+    jupyter = 0
+
+
+class InteractiveSession(Base, Timestamp):
+    """Interactive Session table."""
+
+    __tablename__ = "interactive_session"
```

Comment:
Not to be addressed in this PR:

Addressing the following line in https://github.com/reanahub/reana-db/issues/89:
> (2) opening interactive sessions without necessarily having any workflow yet.

There is now no notion of `workspace` for `InteractiveSessoin`'s, however, the `InteractiveSession`'s implementation depends on it. So far, we have been [depending on the workflow model to retrieve it](https://github.com/reanahub/reana-workflow-controller/blob/master/reana_workflow_controller/workflow_run_manager.py#L287:L287), which was ok because the interactive session was also within the workflow model.

As we are evolving to this new idea of separate sessions, I see a need to [extract the workspace concept from within the `Workflow` model](https://github.com/reanahub/reana-db/blob/8203b6afeb90df9d39c13a4c00e71c43cd0e9077/reana_db/models.py#L328) to a new entity. This idea gains more importance once we start thinking about using different workspaces than CephFS.